rocAL/rocAL/source/readers/image/file_list_reader.cpp

```diff
@@ -0,0 +1,281 @@
+THE SOFTWARE.
+*/
+
+#include <cassert>
+#include <algorithm>
+#include <commons.h>
+#include "file_list_reader.h"
+#include <boost/filesystem.hpp>
+#include <fstream>
+
+
+namespace filesys = boost::filesystem;
+
+FileListReader::FileListReader():
+_shuffle_time("shuffle_time", DBG_TIMING)
+{
+    _src_dir = nullptr;
+    _sub_dir = nullptr;
+    _entity = nullptr;
+    _curr_file_idx = 0;
+    _current_file_size = 0;
+    _current_fPtr = nullptr;
+    _loop = false;
+    _file_id = 0;
+    _shuffle = false;
+    _file_count_all_shards = 0;
+}
+
+unsigned FileListReader::count_items()
+{
+    if(_loop)
+        return _file_names.size();
+
+    int ret = ((int)_file_names.size() -_read_counter);
+    return ((ret < 0) ? 0 : ret);
+}
+
+Reader::Status FileListReader::initialize(ReaderConfig desc)
+{
+    auto ret = Reader::Status::OK;
+    _file_id = 0;
+    _folder_path = desc.path();
+    _file_list_path = desc.json_path();
+    _shard_id = desc.get_shard_id();
+    _shard_count = desc.get_shard_count();
+    _batch_count = desc.get_batch_size();
+    _shuffle = desc.shuffle();
+    _loop = desc.loop();
+    _meta_data_reader = desc.meta_data_reader();
+    ret = subfolder_reading();
+    // the following code is required to make every shard the same size:: required for multi-gpu training
+    if (_shard_count > 1 && _batch_count > 1) {
+        //std::cout << "===SIZE===:" << _file_names.size() << std::endl;
+        int _num_batches = _file_names.size()/_batch_count;
+        int max_batches_per_shard = (_file_count_all_shards + _shard_count-1)/_shard_count;
+        max_batches_per_shard = (max_batches_per_shard + _batch_count-1)/_batch_count;
+        if (_num_batches < max_batches_per_shard) {
+            replicate_last_batch_to_pad_partial_shard();
+        }
+    }
+    //shuffle dataset if set
+    _shuffle_time.start();
+    if( ret==Reader::Status::OK && _shuffle)
+        std::random_shuffle(_file_names.begin(), _file_names.end());
+    _shuffle_time.end();
+    return ret;
+
+}
+
+void FileListReader::incremenet_read_ptr()
+{
+    _read_counter++;
+    _curr_file_idx = (_curr_file_idx + 1) % _file_names.size();
+}
+size_t FileListReader::open()
+{
+    auto file_path = _file_names[_curr_file_idx];// Get next file name
+    // std::cerr<< "\n In Open - file_path "<<file_path;
+    incremenet_read_ptr();
+    _last_file_path = _last_id = file_path;
+    auto last_slash_idx = _last_id.find_last_of("\\/");
+    if (std::string::npos != last_slash_idx)
+    {
+        _last_id.erase(0, last_slash_idx + 1);
+    }
+
+    _current_fPtr = fopen(file_path.c_str(), "rb");// Open the file,
+
+    if(!_current_fPtr) // Check if it is ready for reading
+        return 0;
+
+    fseek(_current_fPtr, 0 , SEEK_END);// Take the file read pointer to the end
+
+    _current_file_size = ftell(_current_fPtr);// Check how many bytes are there between and the current read pointer position (end of the file)
```

Comment:
Add space after ; for the comments for all instances in this file

Reply:
done